src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java

```diff
@@ -0,0 +1,299 @@
+    {
+        @Nullable final Window window = getWindow();
+        if (window != null)
+        {
+            window.open();
+        }
+    }
+
+    /**
+     * Will return the window if this building has an associated BlockOut window.
+     *
+     * @return BlockOut window.
+     */
+    @Nullable
+    public Window getWindow()
+    {
+        return null;
+    }
+
+    /**
+     * Read this view from a {@link ByteBuf}.
+     *
+     * @param buf The buffer to read this view from.
+     */
+    public void deserialize(@NotNull final ByteBuf buf)
+    {
+        buildingLevel = buf.readInt();
+        buildingMaxLevel = buf.readInt();
+        buildingDmPrio = buf.readInt();
+        workOrderLevel = buf.readInt();
+
+        loadRequestSystemFromNBT(ByteBufUtils.readTag(buf));
+    }
+
+    private void loadRequestSystemFromNBT(NBTTagCompound compound)
+    {
+        this.citizensByRequests.clear();
+        if (compound.hasKey(TAG_CITIZEN_BY_REQUEST))
+        {
+            NBTTagList citizensByRequestList = compound.getTagList(TAG_CITIZEN_BY_REQUEST, Constants.NBT.TAG_COMPOUND);
+            NBTUtils.streamCompound(citizensByRequestList).forEach(cbrc -> processIntegerKeyTokenList(cbrc, citizensByRequests));
+        }
+
+
+        this.requestsByCitizen.clear();
+        this.citizensByRequests.keySet().forEach(citizen -> this.citizensByRequests.get(citizen).forEach(requestToken -> this.requestsByCitizen.put(requestToken, citizen)));
+    }
+
+    @SuppressWarnings(Suppression.GENERIC_WILDCARD)
+    public <R> ImmutableList<IRequest<? extends R>> getOpenRequestsOfType(@NotNull final CitizenDataView citizenData, final Class<R> requestType)
+    {
+        return ImmutableList.copyOf(getOpenRequests(citizenData).stream()
+                                      .filter(request -> {
+                                          Set<TypeToken> requestTypes = ReflectionUtils.getSuperClasses(request.getRequestType());
+                                          return requestTypes.contains(requestType);
+                                      })
+                                      .map(request -> (IRequest<? extends R>) request)
+                                      .iterator());
+    }
+
+    public ImmutableList<IRequest> getOpenRequests(@NotNull final CitizenDataView data)
+    {
+        if (!citizensByRequests.containsKey(data.getId()))
+        {
+            return ImmutableList.of();
+        }
+
+        return ImmutableList.copyOf(citizensByRequests.get(data.getId()).stream().map(getColony().getRequestManager()::getRequestForToken).filter(Objects::nonNull).iterator());
+    }
+
+    /**
+     * Gets the ColonyView that this building belongs to.
+     *
+     * @return ColonyView, client side interpretations of Colony.
+     */
+    public ColonyView getColony()
+    {
+        return colony;
+    }
+
+    @SuppressWarnings(Suppression.GENERIC_WILDCARD)
+    public <R> ImmutableList<IRequest<? extends R>> getOpenRequestsOfTypeFiltered(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1452)